File: EntityFrameworkCore.Diagrams/EntityFrameworkCore.Diagrams/Dto/DbEntityForeignKey.cs
﻿using System.Collections.Generic;

namespace EntityFrameworkCore.Diagrams.Dto
{
    public class DbEntityForeignKey : DbEntityForeignKeyBase
    {
        public DbEntity PrincipalEntity { get; internal set; }

        public DbEntityKey PrincipalKey { get; internal set; }

        public IEnumerable<DbEntityProperty> Properties { get; internal set; }
    }
}

File: EntityFrameworkCore.Diagrams/EntityFrameworkCore.Diagrams.Frontend/src/app/models/db-entity-foreign-key.ts
import { DbEntity } from './db-entity';
import { DbEntityKey } from './db-entity-key';
import { DbEntityProperty } from './db-entity-property';

export class DbEntityForeignKey {
    principalEntity: DbEntity;
    principalKey: DbEntityKey;
    properties: DbEntityProperty[] = [];

    static fromJSON(obj: Object): DbEntityForeignKey {
        return Object.assign(new DbEntityForeignKey(), obj, {
            principalEntity: DbEntity.fromJSON(obj['principalEntity']),
            principalKey: DbEntityKey.fromJSON(obj['principalKey']),
            properties: (obj['properties'] || []).map(e => DbEntityProperty.fromJSON(e)),
        });
    }
}
